qlog/qlog.go

```diff
@@ -139,13 +142,15 @@ func (t *tracer) StartedConnection(local, remote net.Addr, version protocol.Vers
 	if !ok {
 		return
 	}
+	t.mutex.Lock()
```

Comment:
If you're doing this every time you call recordEvent, why not move the lock/unlock into that function?

Reply:
The thing this mutex protects is the call to `time.Now()`. We need to ensure that events are ordered when we write them to the qlog.

Reply:
Then why not do the time.Now() within recordEvent?

Reply:
1. Some events produce two log entries. Would be nice to have the same time stamp for the them.
2. `SetLossTimer` uses the difference between the timestamp of the event and the deadline.

Reply:
sgtm